docs/src/main/asciidoc/getting-started-reactive.adoc

```diff
@@ -0,0 +1,854 @@
+This engine is Eclipse Vert.x (https://vertx.io).
+Every IO interaction passes through the non-blocking and reactive Vert.x engine.
+
+image:quarkus-reactive-stack.png[alt=Quarkus is based on a reactive engine, 50%]
+
+Let's take 2 examples to explain how it works.
+Imagine an incoming HTTP request.
+The (Vert.x) HTTP server receives the request and then routes it to the application.
+If the request targets a JAX-RS resource, the routing layer invokes the resource method in a worker thread and writes the response when the data is available.
+So far, nothing new or outstanding.
+The following picture depicts this behavior.
+
+image:http-blocking-sequence.png[alt=Behavior when using the imperative routes, 50%]
+
+But if the HTTP request targets a reactive (non-blocking) route, the routing layer invokes the route on the IO thread giving lots of benefits such as higher concurrency and performance:
+
+image:http-reactive-sequence.png[alt=Behavior when using the reactive routes, 50%]
+
+As a consequence, many Quarkus components are designed with reactive in mind, such as database access (PostgreSQL, MySQL, Mongo, etc.), application services (mail, template engine, etc.), messaging (Kafka, AMQP, etc.) and so on.
+But, to fully benefit from this model, the application code should be written in a non-blocking manner.
+That’s where having a reactive API is an ultimate weapon.
+
+== Mutiny - A reactive programming library
+
+https://github.com/smallrye/smallrye-mutiny[Mutiny] is a reactive programming library allowing to express and compose asynchronous actions.
+It offers 2 types:
+
+* `io.smallrye.mutiny.Uni` - for asynchronous action providing 0 or 1 result
+* `io.smallrye.mutiny.Multi` - for multi-item (with back-pressure) streams
+
+Both types are lazy and follow a subscription pattern.
+The computation only starts once there is an actual need for it (i.e., a subscriber enlists).
+
+[source, java]
+----
+uni.subscribe().with(
+    result -> System.out.println("result is " + result),
+    failure -> failure.printStackTrace()
+);
+
+multi.subscribe().with(
+    item -> System.out.println("Got " + item),
+    failure -> failure.printStackTrace()
+);
+----
+
+Both `Uni` and `Multi` expose event-driven APIs: you express what you want to do upon a given event (success, failure, etc.).
+These APIs are divided into groups (types of operations) to make it more expressive and avoid having 100s of methods attached to a single class.
+The main types of operations are about reacting to failure, completion, manipulating items, extracting, or collecting them.
+It provides a smooth coding experience, with a navigable API, and the result does not require too much knowledge around reactive.
+
+[source, java]
+----
+httpCall
+    .onFailure().recoverWithItem("my fallback");
+----
+
+You may wonder about Reactive Streams (https://www.reactive-streams.org/).
+`Multi` implements Reactive Streams `Publisher`, and so implements the Reactive Streams back-pressure mechanism.
```

Comment:
Aren't we going to ditch that at some point?

Reply:
Publisher is the base, we removed Reactive Streams Operators.